app/validation.py

```diff
@@ -41,8 +42,14 @@ def load_schemas(schemas_path, schema_names):
 _SCHEMAS = load_schemas(JSON_SCHEMAS_PATH, SCHEMA_NAMES)
 
 
-def get_validator(schema_name):
-    schema = _SCHEMAS[schema_name]
+def get_validator(schema_name, enforce_required=True, required_fields=None):
+    if required_fields is None:
+        required_fields = []
+    if enforce_required:
+        schema = _SCHEMAS[schema_name]
+    else:
+        schema = _SCHEMAS[schema_name].copy()
```

Comment:
The only reason we need `.copy` here is to avoid modifying the original schema. The copied object still has all the nested elements, but they're shared with the original. `.deepcopy` would create copies of the nested structures, but since we're not actually modifying any of them there's no need for us to do this.